packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -1115,37 +1102,48 @@ abstract class FlameChartPainter extends CustomPainter {
   FlameChartPainter({
     @required this.zoom,
     @required this.constraints,
-    @required this.verticalScrollOffset,
-    @required this.horizontalScrollOffset,
+    @required this.verticalScroll,
+    @required this.horizontalScroll,
     @required this.chartStartInset,
     @required this.colorScheme,
-  })  : visible = Rect.fromLTWH(
-          horizontalScrollOffset,
-          verticalScrollOffset,
-          constraints.maxWidth,
-          constraints.maxHeight,
-        ),
-        assert(colorScheme != null);
+  })  : assert(colorScheme != null),
+        super(
+            repaint: Listenable.merge([
+          verticalScroll,
+          horizontalScroll.offsetNotifier,
+        ]));
 
   final double zoom;
 
   final BoxConstraints constraints;
 
-  final double verticalScrollOffset;
+  final ScrollController verticalScroll;
 
-  final double horizontalScrollOffset;
+  final LinkedScrollControllerGroup horizontalScroll;
```

Comment:
nit: let's name these with controller and controllerGroup at the end to make it clear that these are not offsets, or in the case of horizontalScroll, not even a scroll controller.

Reply:
In general you should avoid including types in member names as the language is typed so there is no need to specify it. However,  I do agree that clarifying those are controllers is a good idea.
how about `horizontalController` and `verticalController` so that we avoid overly verbose Java style variable names. Some prior art from flutter is the ScrollController for ListView is just called `controller` not `scrollController` and the `child` property is `child` not `childWidget`.